lib/trino-parquet/src/main/java/io/trino/parquet/reader/decoders/ValueDecoders.java

```diff
@@ -167,6 +174,24 @@ private static ValueDecoder<long[]> getBinaryLongDecimalDecoder(ParquetEncoding
         };
     }
 
+    public static ValueDecoder<BinaryBuffer> getBinaryDecoder(ParquetEncoding encoding, PrimitiveField field, @Nullable Dictionary dictionary)
+    {
+        return switch (encoding) {
+            case PLAIN, DELTA_LENGTH_BYTE_ARRAY, DELTA_BYTE_ARRAY, PLAIN_DICTIONARY, RLE_DICTIONARY -> {
+                ValuesReader delegate = getApacheParquetReader(encoding, field, dictionary);
+                Type trinoType = field.getType();
+                if (trinoType instanceof VarcharType varcharType && !varcharType.isUnbounded()) {
```

Comment:
I actually meant for this to be in the ColumnReaderFactory in the comment I made earlier. This method should not be switching on Trino types.

Reply:
Right, fixed this now